kernel/kernel-default/src/test/java/io/delta/kernel/client/TestDefaultJsonHandler.java

```diff
@@ -0,0 +1,181 @@
+            "part-00001-e62ca5a1-923c-4ee6-998b-c61d1cfb0b1c-c000.snappy.parquet"
+        );
+        List<Long> expSizes = Arrays.asList(348L, 687L, 705L, 650L, 650L, 649L, 649L);
+        List<Boolean> expDataChanges = Arrays.asList(true, true, true, true, true, true, true);
+
+        assertEquals(expPaths, actPaths);
+        assertEquals(expSizes, actSizes);
+        assertEquals(actDataChanges, expDataChanges);
+    }
+
+    @Test
+    public void parseJsonContent()
+        throws Exception
+    {
+        String input =
+            "{" +
+                "   \"path\":\"part-00000-d83dafd8-c344-49f0-ab1c-acd944e32493-c000.snappy.parquet\", " +
+                "   \"partitionValues\":{\"p1\" : \"0\", \"p2\" : \"str\"}," +
+                "   \"size\":348," +
+                "   \"modificationTime\":1603723974000, " +
+                "   \"dataChange\":true" +
+                "   }";
+        StructType readSchema = new StructType()
+            .add("path", StringType.INSTANCE)
+            .add("partitionValues",
+                new MapType(StringType.INSTANCE, StringType.INSTANCE, false))
+            .add("size", LongType.INSTANCE)
+            .add("dataChange", BooleanType.INSTANCE);
+
+        ColumnarBatch batch =
+            JSON_HANDLER.parseJson(Utils.singletonColumnVector(input), readSchema);
+        assertEquals(1, batch.getSize());
+
+        try (CloseableIterator<Row> rows = batch.getRows()) {
+            Row row = rows.next();
+            assertEquals(
+                "part-00000-d83dafd8-c344-49f0-ab1c-acd944e32493-c000.snappy.parquet",
+                row.getString(0)
+            );
+
+            Map<String, String> expPartitionValues = new HashMap<String, String>() {{
+                put("p1", "0");
+                put("p2", "str");
+            }};
+            assertEquals(expPartitionValues, row.getMap(1));
+            assertEquals(348L, row.getLong(2));
+            assertEquals(true, row.getBoolean(3));
+        }
+    }
+
+    private static CloseableIterator<Row> testFiles()
+        throws Exception
+    {
+        String listFrom = getTestResourceFilePath("json-files/1.json");
+        CloseableIterator<FileStatus> list = FS_CLIENT.listFrom(listFrom);
+        return list.map(fileStatus ->
+            new DefaultJsonRow(
+                addFileJsonFromPath(fileStatus.getPath()),
+                new StructType()
+                    .add("path", StringType.INSTANCE)
+                    .add("dataChange", BooleanType.INSTANCE)
+                    .add("size", LongType.INSTANCE)
+            )
+        );
+    }
+
+    private static final ObjectNode addFileJsonFromPath(String path)
+    {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode object = objectMapper.createObjectNode();
+        object.put("path", path);
+        object.put("dataChange", true);
+        object.put("size", 234L);
+        return object;
+    }
```

Comment:
Aren't we adding row implementations at some point that could be used for this? Can we just add it in this PR and use it here? This is needlessly confusing

Reply:
I am not aware of plans for adding Row implementations. Even if we add those will be mostly around the wrappers around the columnarbatch.

What is confusing? Using the DefaultJsonRow here?

Reply:
We're creating an `ObjectNode` to create a `JsonRow` when all all we really need is the Row